packages/adapter/src/lib/adapter/adapter.js

```diff
@@ -8369,28 +8370,58 @@ function Adapter(options) {
                 adapterStates.setBinaryState(id, binary, callback);
             }
         };
+
         /**
          * Promise-version of Adapter.setBinaryState
          *
-         * @alias setBinaryStateAsync
+         * @alias setForeignBinaryStateAsync
          * @memberof Adapter
          * @param {string} id of state
          * @param {Buffer} binary data
          * @param {object} [options] optional
-         * @return promise
+         * @return {Promise}
+         *
+         */
+        this.setForeignBinaryStateAsync = tools.promisify(this.setForeignBinaryState, this);
+
+        /**
+         * Same as setForeignBinaryState but calls fixId first
```

Comment:
Nit: I don't think "calls fixId first" is helpful information in the syntax help. Maybe 
```suggestion
         * Same as setForeignBinaryState but automatically prepends the adapter namespace if necessary
```